controls/views.py

```diff
@@ -3591,4 +3595,52 @@ def new_system_assessment_result_wazuh(request, system_id):
 
         else:
             # TODO: better handling of response code; 401, 301, etc.
-            raise Exception(wazuh_sec_svc.error_msg['error'])
\ No newline at end of file
+            raise Exception(wazuh_sec_svc.error_msg['error'])
+
+
+
+@login_required
+def update_control_status(request, system_id):
+    """Update the control status"""
+    if request.method != "POST":
+        return HttpResponseNotAllowed(["POST"])
+    else:
+        form_dict = dict(request.POST)
+
+        eid = form_dict["eid"][0]
+        status = form_dict["status"][0]
+
+        control = ElementControl.objects.get(id=eid)
+        control.status = status
+
+        system = System.objects.get(id=system_id)
+
+        if not request.user.has_perm('change_system', system):
+            # User does not have write permissions
+            # Log permission to save answer denied
+            logger.info(
+                event="update_control_status permission_denied",
+                object={"object": "control", "id": control.id},
+                user={"id": request.user.id, "username": request.user.username}
+            )
+            return HttpResponseForbidden(
+                "Permission denied. {} does not have change privileges to system and/or project.".format(request.user.username))
+
+        if control is None:
+            control_msg = "The id for this control is no longer valid in the database."
+            return JsonResponse({"message": control_msg}, status=422)
+
+        allowed = ["2","3","4","5"]
```

Comment:
Can we pull these values out of ElementControl.Statuses so there is only one source of truth about what the allowed values are?